packages/frontend/src/page/index.tsx

```diff
@@ -65,7 +93,8 @@ export const Component = ({
     useEffect(() => {
         let cancelled = false;
         const fetchData = async () => {
-            if (!kpiToken || !kpiTokenData) return;
+            if (!kpiToken || !kpiTokenData || !publicClient || !tokenData)
```

Comment:
Here I'd personally split the hook in 3 parts. The first one should save to the component's state a `Token` version of the KPI token based on its name, symbol and decimals. With that saved version we can then have 2 additional hooks that fetch the decoded data and update the current supply respectively.